src/main/scala/viper/gobra/backend/SilFrontendForFrontends.scala

```diff
@@ -0,0 +1,87 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2021 ETH Zurich.
+
+package viper.gobra.backend
+
+import java.nio.file.Path
+
+import viper.silicon
+import viper.silver.ast.{Node, Program}
+import viper.silver.frontend.{DefaultStates, SilFrontend}
+import viper.silver.verifier.ConsistencyError
+
+/**
+  * This trait adapts a SilFrontend to Viper frontends that encode a program into a Viper program and thus do not need
+  * execute phases related to parsing. Key to use this trait is setting `encoding` to the Viper program that the
+  * frontend has produced.
+  * `execute` should be used to run the phases.
```

Comment:
```suggestion
  * This trait adapts the SilFrontend so that frontends that encode a program into a Viper program do not need to perform parsing. 
  * An implementation of this trait has to overwrite the `encoding` field with targeted Viper AST. 
  * To verify a program, create an instance, `execute`, and read the result form the `result` field. 
  * Careful: `execute` must be called as most once.
```

Is the last sentence correct? I yes, then can we change this? e.g. either check that execute is only called once.